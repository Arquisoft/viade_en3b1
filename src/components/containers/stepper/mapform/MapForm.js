import React, { Component } from 'react';
import EditableMap from '../../../map/EditableMap';
import { Button, Typography, Grid, Snackbar, IconButton } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import MuiAlert from '@material-ui/lab/Alert';

export class MapForm extends Component {

    constructor() {
        super();
        this.points = React.createRef();
        this.state = {
            open: false,
            message: "",
            severity: '',
            vertical: 'top',
            horizontal: 'center',
        };
    }

    next = (e) => {
        e.preventDefault();
        if (this.points.current.getTrackPoints() === 'undefined' || this.points.current.getTrackPoints().length === 0) {
            this.openNotif("You must select at least one track point!!", 'warning');
            return;
        }
        this.props.handleNext();
        this.props.handleMapPoints(this.points.current.getTrackPoints());
    }

    back = (e) => {
        e.preventDefault();
        this.props.handleBack();
    }

    openNotif = (text, newSeverity) => {
        this.setState({
            open: true,
            message: text,
            severity: newSeverity
        });
    };

    closeNotif = () => {
        this.setState({ open: false });
    };

    render() {
        const { open, message, severity } = this.state;
        const { vertical, horizontal } = this.state;

        return (
            <React.Fragment>
                <Snackbar
                    anchorOrigin={{ vertical, horizontal }}
                    open={open}
                    action={
                        <React.Fragment>
                            <IconButton
                                aria-label="close"
                                color="inherit"
                                onClick={this.closeNotif}
                            >
                                <CloseIcon />
                            </IconButton>
                        </React.Fragment>
                    }
                >
                    <Alert onClose={this.closeNotif} severity={severity}>
                        {message}
                    </Alert>
                </Snackbar>

                <Typography variant="h6" gutterBottom>
                    Click on the map to add trackpoints to your route
                </Typography>

                <form onSubmit={this.next}>
                    <EditableMap ref={this.points} openNotif={this.openNotif} />

                    <Grid container spacing={3} style={{ marginTop: 12 }}>
                        <Button
                            variant="contained"
                            color="default"
                            style={{
                                marginTop: 12,
                                marginLeft: '1vh'
                            }}
                            type="submit"
                            onClick={this.back}
                        >
                            Back
                        </Button>

                        <Button
                            variant="contained"
                            color="primary"
                            style={{
                                marginTop: 12,
                                marginLeft: 'auto'
                            }}
                            type="submit"
                        >
                            Next
                        </Button>
                    </Grid>
                </form>
            </React.Fragment>
        );
    }
}

function Alert(props) {
    return <MuiAlert elevation={6} variant="filled" {...props} />;
}

export default MapForm;

