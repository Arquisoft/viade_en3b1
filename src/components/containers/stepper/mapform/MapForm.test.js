import MapForm from './MapForm';

const state = {
  activeStep: 0,
  name: '',
  description: '',
  date: new Date(),
  photos: [],
  videos: [],
  points: []
}

const { activeStep, name, description, date, photos, videos, points } = state;
const values = { activeStep, name, description, date, photos, videos, points };

const handleNext = jest.fn();
const handleBack = jest.fn();
const handleChange = jest.fn();
const handleDateChange = jest.fn();
const handleMediaChange = jest.fn();
const handleMapPoints = jest.fn();
const handleDownload = jest.fn();
const createRoute = jest.fn();

test('Define MapForm', () => {
  expect(MapForm).toBeDefined();
});

test('Renders correctly', () => {
  const wrapper = render(<MapForm
    handleNext={handleNext}
    handleBack={handleBack}
    handleMapPoints={handleMapPoints}
    values={values}
/>);
  expect(wrapper).toMatchSnapshot();
});