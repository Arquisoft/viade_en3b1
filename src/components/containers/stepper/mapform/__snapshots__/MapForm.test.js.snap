// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Array [
  <h6
    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
  >
    Click on the map to add trackpoints to your route
  </h6>,
  <form>
    <div
      id="test-map-leaflet"
      style="width:34.5rem;height:20rem"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      style="margin-top:12px"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        style="margin-top:12px;margin-left:1vh"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Back
        </span>
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        style="margin-top:12px;margin-left:auto"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Next
        </span>
      </button>
    </div>
  </form>,
]
`;
